lib/jxl/dec_cache.cc

```diff
@@ -195,23 +196,57 @@ Status PassesDecoderState::PreparePipeline(ImageBundle* decoded,
       }
     }
 
+    ColorEncoding linear_color_encoding;
+    if (frame_header.color_transform == ColorTransform::kXYB) {
+      linear_color_encoding = output_encoding_info.linear_color_encoding;
+    } else {
+      linear_color_encoding = ColorEncoding::LinearSRGB(
```

Comment:
I admit I don't exactly remember the semantics of linear_color_encoding, but is that supposed to change primaries wrt the original ones? I'd guess not...

Reply:
Yes, my point is that that linear colorspace will likely not have SRGB primaries if the data encoding did not have SRGB primaries...

Reply:
Is that true, @sboukortt? Does it have the primaries of the data encoding or always SRGB primaries?

Reply:
After some consideration, we figured out (probably...) that we don't need that variable anymore.

Reply:
Yes, removed it. For completeness: `linear_color_encoding` should be in the primaries of the ouput color space.